mllib/src/main/scala/org/apache/spark/ml/stat/ChiSquareTest.scala

```diff
@@ -75,4 +77,31 @@ object ChiSquareTest {
     val statistics: Vector = Vectors.dense(testResults.map(_.statistic))
     spark.createDataFrame(Seq(ChiSquareResult(pValues, degreesOfFreedom, statistics)))
   }
+
+  /**
+   * @param dataset  DataFrame of categorical labels and categorical features.
+   *                 Real-valued features will be treated as categorical for each distinct value.
+   * @param featuresCol  Name of features column in dataset, of type `Vector` (`VectorUDT`)
+   * @param labelCol  Name of label column in dataset, of any numerical type
+   * @return Array containing the SelectionTestResult for every feature against the label.
+   */
+  @Since("3.1.0")
```

Comment:
That difference between this method and above `def test(dataset: DataFrame, featuresCol: String, labelCol: String): DataFrame` is only that the result are parallelized to a dataframe?

Reply:
Yes. 
I may need to refactor this method and move it out of this class in next subtask.